dbt_coves/templates/source_model.sql

```diff
@@ -12,19 +12,19 @@ final as (
 {%- if adapter_name == 'SnowflakeAdapter' %}
 {%- for key, cols in nested.items() %}
   {%- for col in cols %}
-        {{ key }}:{{ '"' + col + '"' }}::varchar as {{ col.lower().replace(" ","_").replace(":","_").replace("(","_").replace(")","_").replace("-","_").replace("/","_") }}{% if not loop.last or columns %},{% endif %}
+        {{ key }}:{{ '"' + col + '"' }}::{{ cols[col]["type"] }} as {{ col.lower().replace(" ","_").replace(":","_").replace("(","_").replace(")","_").replace("-","_").replace("/","_") }}{% if not loop.last or columns %},{% endif %}
```

Comment:
It'd be great if we could replace `col.lower().replace(" ","_").replace(":","_").replace("(","_").replace(")","_").replace("-","_").replace("/","_")` by `cols[col]["id"]` where that `id` contains a valid id string (we can use https://github.com/un33k/python-slugify with "_" as separator).

Reply:
@ssassi  done